test/unit/master/test_cluster_master.py

```diff
@@ -213,6 +219,40 @@ def test_updating_slave_to_nonexistent_state_should_raise_bad_request_error(self
         with self.assertRaises(BadRequestError):
             master.handle_slave_state_update(slave, 'NONEXISTENT_STATE')
 
+    def test_update_slave_last_heartbeat_time_calls_update_last_heartbeat_time_on_slave(self):
+        master = ClusterMaster()
+
+        mock_slave = self.patch('app.master.cluster_master.Slave').return_value
+        # self.patch('app.master.cluster_master.Slave', new=lambda *args: mock_slave)
+        # master.connect_slave('slave_url', 1)
+        master.update_slave_last_heartbeat_time(mock_slave)
+
+        self.assertEqual(mock_slave.update_last_heartbeat_time.call_count, 1,
+                         'last heartbeat time is updated for the target slave')
+
+    @genty_dataset (
+        slave_unresponsive=(True,1000,),
+        slave_dead=(False,60,),
+        slave_responsive=(True,60,),
+    )
+    def test_heartbeat_disconnects_unresponsive_slave(self, slave_alive, seconds_since_last_heartbeat):
+        last_heartbeat_time = self._mock_current_datetime - timedelta(seconds=seconds_since_last_heartbeat)
+        master = ClusterMaster()
+
+        mock_slave = Mock()
```

Comment:
Me and Evan ran into a lot of issues in mocking this particular class. If I do specify `spec` I get the following error. It is because of the way we use public member id. I proper fix may be to change `id` to a property, but not sure if that should be a part of this PR:

```
AttributeError: Mock object has no attribute 'id'
```